plan/predicate_push_down.go

```diff
@@ -154,6 +154,12 @@ func (p *LogicalJoin) PredicatePushDown(predicates []expression.Expression) (ret
 		p.LeftJoinKeys = append(p.LeftJoinKeys, eqCond.GetArgs()[0].(*expression.Column))
 		p.RightJoinKeys = append(p.RightJoinKeys, eqCond.GetArgs()[1].(*expression.Column))
 	}
+	// After predicate push down, the size of schema may change, so we should also change the len of default value.
+	if p.JoinType == LeftOuterJoin {
```

Comment:
Can we do it lazily?
Like build the default value in executor?

Reply:
If we enable agg push down. The default value cannot be decided easily if we don't decide it when pushing agg down.

Reply:
I think it's not safe to overwrite the old default values.

Reply:
Another way is only set default values in agg push down. So if there is no agg push down, default values is nil, and in executor phase we can generate a slice to build the default values. If there is agg push down, the default values is not nil. In executor phase we just use it.

Reply:
It's safe because agg push down is after the phase that may change the schema len.

Reply:
If the default value is not all null like aggregation which will produce not null default value.
This will overwrite it to null.

Reply:
@coocood This only happens in agg push down which is after the predicate pushed after we enable it.

Reply:
@winoros I find that the default values are always NULLs before we do the "aggregate push down" optimization, and only in that phase default values may changed to not NULL values.

Maybe we can do as you said: let `DefaultValues` be nil and only set it during the "aggregate push down" stage, by doing this, we no need to maintain `DefaultValues` before "aggregate push down"

But I think it's better to set `DefaultValues` after physical optimization, this work is part of plan building, better not to be done in executor building stage.

Reply:
And we can remove `resetDefaultValues()` in projection elimination.

Reply:
If we do so, the `join.DefaultValues` is nil just means that the default values of this join is a slice of nil. So we dicide the values in plan phase actually.